src/uu/nohup/src/nohup.rs

```diff
@@ -123,46 +167,39 @@ fn replace_fds() -> UResult<()> {
     Ok(())
 }
 
-fn find_stdout() -> UResult<File> {
-    let internal_failure_code = match env::var("POSIXLY_CORRECT") {
-        Ok(_) => POSIX_NOHUP_FAILURE,
-        Err(_) => EXIT_CANCELED,
-    };
+/// Open nohup.out file with mode 0o600, temporarily clearing umask.
+/// The umask is cleared to ensure the file is created with exactly 0o600 permissions.
+fn open_nohup_file(path: &Path) -> std::io::Result<File> {
+    // Clear umask (set it to 0) and save the old value
+    let old_umask = umask(Mode::from_bits_truncate(0));
 
-    match OpenOptions::new()
+    let result = OpenOptions::new()
         .create(true)
         .append(true)
-        .open(Path::new(NOHUP_OUT))
-    {
-        Ok(t) => {
```

Comment:
When implementing this, it doesn't appear that this matches the GNU implementation, I have added integration tests here and this is not something I came across when trying to replicate the GNU functionality